snorkel/labeling/lf/nlp.py

```diff
@@ -0,0 +1,202 @@
+from typing import Any, Callable, List, Mapping, NamedTuple, Optional
+
+from snorkel.labeling.preprocess import BasePreprocessor
+from snorkel.labeling.preprocess.nlp import EN_CORE_WEB_SM, SpacyPreprocessor
+
+from .core import LabelingFunction
+
+
+class SpacyPreprocessorParameters(NamedTuple):
+    """Parameters need to construct a SpacyPreprocessor."""
+
+    text_field: str
+    doc_field: str
+    language: str
+    disable: Optional[List[str]]
+    preprocessors: List[BasePreprocessor]
+    memoize: bool
+
+
+class SpacyPreprocessorConfig(NamedTuple):
+    """Tuple of SpacyPreprocessor and the parameters used to construct it."""
+
+    nlp: SpacyPreprocessor
+    parameters: SpacyPreprocessorParameters
```

Comment:
Seems like defining this isn't really necessary?  Seems a little overkill?

Reply:
The tuple? It's a cleaner way to enforce mutual inclusivity and other properties of the parameters than a bunch of logical checks.

Reply:
This is not major but looking down below, I still think it's confusing that the "config" object contains the actual preprocessor